app/web/template/layout/page.gohtml

```diff
@@ -96,13 +96,13 @@
 
           <ul class="trans-switch" id="language-switch" aria-label="Change your language">
             <li class="govuk-list trans-switch__list govuk-body-s" id="language-switch-en" lang="en-gb">
-              {{ if isWelsh .App.Lang }}<a href="{{ .App.Page }}" class="govuk-link"><span class="govuk-visually-hidden">Change language to </span>{{ end }}
+              {{ if isWelsh .App.Lang }}<a href="{{ join "" .App.Page .App.Query }}" class="govuk-link"><span class="govuk-visually-hidden">Change language to </span>{{ end }}
                 {{ if isEnglish .App.Lang }}<span class="govuk-visually-hidden">Current language is </span>{{ end }}
                 English
                 {{ if isWelsh .App.Lang }}</a>{{ end }}
             </li>
             <li class="govuk-list trans-switch__list govuk-body-s" id="language-switch-cy" lang="cy-gb">
-              {{ if isEnglish .App.Lang }}<a href="/cy{{ .App.Page }}" class="govuk-link" aria-label="Newid iaith i'r Gymraeg">{{ end }}
+              {{ if isEnglish .App.Lang }}<a href="/cy{{ join "" .App.Page .App.Query }}" class="govuk-link" aria-label="Newid iaith i'r Gymraeg">{{ end }}
```

Comment:
could just have done `<a href="/cy{{ .App.Page }}{{ .App.Query }}" ...`

Reply:
Haha that's hilarious - always reaching for a sledgehammer to crack a walnut. I'll banish `join()` back to the depths.